cpp/farm_ng/core/prototools/event_log_writer_test.cpp

```diff
@@ -44,15 +44,140 @@ TEST(event_log, roundtrip) {
     EventLogReader reader(file);
     for (int i = 0; i < 10; ++i) {
       std::string payload;
-      auto [event, pos] = reader.readNextEvent(&payload);
+      EventLogPos pos = reader.readNextEvent(&payload);
       core::proto::Timestamp x;
       EXPECT_EQ(true, x.ParseFromString(payload));
       EXPECT_EQ(i, x.stamp());
-      EXPECT_EQ("protobuf", event.uri().scheme());
-      EXPECT_EQ("my_stamps", event.uri().path());
-      EXPECT_EQ(getHostName(), event.uri().authority());
-      EXPECT_EQ("type=farm_ng.core.proto.Timestamp", event.uri().query());
+      EXPECT_EQ("protobuf", pos.event().uri().scheme());
+      EXPECT_EQ("my_stamps", pos.event().uri().path());
+      EXPECT_EQ(getHostName(), pos.event().uri().authority());
+      EXPECT_EQ("type=farm_ng.core.proto.Timestamp", pos.event().uri().query());
     }
     EXPECT_THROW(reader.readNextEvent(), std::runtime_error);
   }
 }
+
+TEST(event_log, index) {
+  auto maybe_log_dir = createUniqueTemporaryDirectory();
+  auto log_dir = FARM_UNWRAP(maybe_log_dir);
+  auto file = log_dir / "events.log";
+  {
+    EventLogWriter writer(file);
+    core::proto::Timestamp x;
+    for (int i = 0; i < 10; ++i) {
+      x.set_stamp(i);
+      writer.write("my_stamps", x);
+    }
+  }
+  {
+    EventLogReader reader(file);
+    int i = 0;
+    for (EventLogPos const& pos : reader.getIndex()) {
```

Comment:
@strasdat this api does not throw on file reading,  as it seeks and builds up an index first.  I'll add an example with no index which does throw.